python/feldera/_helpers.py

```diff
@@ -15,14 +56,37 @@ def ensure_dataframe_has_columns(df: pd.DataFrame):
         )
 
 
-def dataframe_from_response(buffer: list[list[dict]]):
+def dataframe_from_response(buffer: list[list[dict]], schema: dict):
     """
     Converts the response from Feldera to a pandas DataFrame.
     """
-    return pd.DataFrame([
+
+    pd_schema = {}
+
+    decimal_col = []
+
+    for column in schema['fields']:
+        column_name = column['name']
+        column_type = column['columntype']['type']
+        if column_type == 'DECIMAL':
+            decimal_col.append(column_name)
+
+        pd_schema[column_name] = sql_type_to_pandas_type(column_type)
+
+    data = [
         {**item['insert'], 'insert_delete': 1} if 'insert' in item else {**item['delete'], 'insert_delete': -1}
         for sublist in buffer for item in sublist
-    ])
+    ]
+
+    for datum in data:
```

Comment:
is it a scalable solution to create the dataframe in one shot by processing all the data?

Reply:
Here, the data only contains the one chunk of many chunks streamed by the backend. One chunk can contain multiple rows but the size is limited by the backend (if I remember correctly). 

So, it shouldn't be a lot of rows, which should be fine to process this way.